ard_controller/ard_controller.ino

```diff
@@ -1,12 +1,22 @@
 #include <DueTimer.h>
 
+#define NUM_JOINTS 8
+#define NUM_IMU_AXES 3
 #define PREAMBLE_LENGTH 4
 #define DATA_BYTE_LENGTH 2
 
 enum SerialReadState {
   INIT,
-  PREAMBLE,
-  
+  READ_PREAMBLE,
+  READ_L_HIP,
+  READ_L_KNEE,
+  READ_R_HIP,
+  READ_R_KNEE,
+  READ_L_SHOULDER,
+  READ_L_ELBOW,
+  READ_R_SHOULDER,
+  READ_R_ELBOW,
+  READ_CHECKSUM
```

Comment:
I'm assuming that an analog for this will have to be implemented on the python/ros side?

Reply:
On the python/ros side, it just has to send the bytes in order, so no need for a full on state machine. You only need a state machine for reading in serial data (which does exist on the python/ros side still for the sensor data service).